orangecontrib/prototypes/widgets/owimagegrid.py

```diff
@@ -0,0 +1,1141 @@
+            self.__grid.setMinimumHeight(height)
+
+    def keyPressEvent(self, event):
+        if event.key() == Qt.Key_Escape and self.__previewWidget is not None:
+            self.__closePreview()
+            event.accept()
+            return
+        return super().keyPressEvent(event)
+
+    def __previewToogle(self):
+        if self.__previewWidget is None and self.__grid.currentItem() is not None:
+            focusitem = self.__grid.currentItem()
+            preview = self.__getPreviewWidget()
+            preview.show()
+            preview.raise_()
+            preview.setPixmap(focusitem.pixmap())
+        else:
+            self.__closePreview()
+
+    def __getPreviewWidget(self):
+        # return the preview image view widget
+        if self.__previewWidget is None:
+            self.__previewWidget = Preview(self)
+            self.__previewWidget.setWindowFlags(
+                Qt.WindowStaysOnTopHint | Qt.Tool)
+            self.__previewWidget.setAttribute(
+                Qt.WA_ShowWithoutActivating)
+            self.__previewWidget.setFocusPolicy(Qt.NoFocus)
+            self.__previewWidget.installEventFilter(self)
+
+        return self.__previewWidget
+
+    def __updatePreviewPixmap(self):
+        current = self.__grid.currentItem()
+        if isinstance(current, GraphicsThumbnailWidget) and \
+                        current.parentItem() is self.__grid and \
+                        self.__previewWidget is not None:
+            self.__previewWidget.setPixmap(current.pixmap())
+
+    def __closePreview(self):
+        if self.__previewWidget is not None:
+            self.__previewWidget.close()
+            self.__previewWidget.setPixmap(QPixmap())
+            self.__previewWidget.deleteLater()
+            self.__previewWidget = None
+
+    def eventFilter(self, receiver, event):
+        if receiver is self.__previewWidget and \
+                        event.type() == QEvent.KeyPress:
+            if event.key() in [Qt.Key_Left, Qt.Key_Right,
+                               Qt.Key_Down, Qt.Key_Up]:
+                self.__grid._moveCurrent(event.key())
+                event.accept()
+                return True
+            elif event.key() in [Qt.Key_Escape, Qt.Key_Space]:
+                self.__closePreview()
+                event.accept()
+                return True
+        return super().eventFilter(receiver, event)
+
+    def hideEvent(self, event):
+        super().hideEvent(event)
+        self.__closePreview()
+
+    def __onCurrentThumbnailChanged(self, thumbnail):
+        if thumbnail is not None:
+            self.__updatePreviewPixmap()
+        else:
+            self.__closePreview()
+
+    @Slot(QPointF)
+    def __ensureVisible(self, point):
+        self.ensureVisible(QRectF(point, QSizeF(1, 1)), 5, 5),
+
+
+# TODO only loads when workdir is image dir
+def main(argv=None):
+    import sys
+    """
+    Will be uncommented when the widget is moved to ImageAnalytics.
```

Comment:
Mark future work with 'TODO' so it is more easily spotted / IDE notifies about.

Reply:
Also why? If running the prototype file will work, it'll work; if it won't, it won't. With this commented out, it just never will.

Reply:
I couldn't figure out a way to make the imports for ImportImages and ImageEmbedder work since they reside in the imageanalytics module. Pycharm did not help me either.

Reply:
I don't understand. Imports as you have them are correct. When imageanalytics add-on is properly installed (i.e. `pip install -e` if prototypes was installed with `-e`), they will resolve. In either event, what good is a main function that just returns?

Reply:
Consider constructing `main()` in such a way that someone who's a working installation of orange3-imageanalytics can run the module and see what's what-

Reply:
The main function now runs properly. Looks like I had to reinstall ImageAnalytics.